GCS: Add board specific ADC pin names

tracernz opened the pull request:
Board plugins can now provide names for the ADC pins. The way this is implemented is slightly hacky (it just replaces the combobox text set by the UAVO->widget relation) but I think it's the cleanest way I could accomplish this without having to manipulate/update FlightBatterySettings by hand. The usability win is large (this has been a pet hate of mine since I first began using Tau Labs).

Naze32 Example:
![screenshot from 2016-01-16 00 50 58](https://cloud.githubusercontent.com/assets/9995998/12352496/60c597b8-bbeb-11e5-8ee1-b0685e18dac0.png)

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/d-ronin/dronin/449)

<!-- Reviewable:end -->

tracernz committed:
GCS:Config: Use data to match combobox items instead of text

This allows the text to be changed by other code without breaking the UAVO->widget relation.

tracernz committed:
Config:Modules: Allow board-specific ADC pin names

tracernz committed:
GCS:Naze32: Add ADC pin names

tracernz committed:
GCS:Sparky: Add ADC pin names

tracernz committed:
GCS:Lux: Add ADC pin names

tracernz committed:
GCS:AQ32: Add ADC pin names

tracernz committed:
GCS:Colibri: Add ADC pin names

tracernz committed:
GCS:Quanton: Add ADC pin names

tracernz committed:
GCS:Revo: Add ADC pin names

tracernz committed:
GCS:Sparky2: Add ADC pin names

tracernz committed:
GCS:Brain: Add ADC pin names

mluessi commented:
Awsome! Is it possible to also use the board specific name in the RSSI channel selection when analog RSSI is selected?

tracernz commented:
Yeah, I was thinking about that just now, will check it out.

jihlein commented on ground/gcs/src/plugins/boards_aeroquad/aq32.cpp:
```diff
@@ -139,3 +139,20 @@ int AQ32::queryMaxGyroRate()
+
+    HwAQ32::DataFields settings = hwAQ32->getData();
+    if (settings.ADCInputs == HwAQ32::ADCINPUTS_ENABLED) {
+        return QStringList() << "BM" << "Analog ADC2" << "Analog ADC4";
```
This a great idea.  Can we can we change Analog ADC2 and Analog ADC4 to AI2 and AI4 respectively to match the board silk screens.

tracernz replied:
Yep, that's a mistake on my part. I ended up taking the labels from here instead of the silk: http://aeroquad.com/showwiki.php?title=AeroQuad32-Flight-Control-Board-v2#AeroQuad32-v2-board-pin-assignments

tracernz commented:
I found the RSSI channel groups on the Input page are subtly broken. The code for the channel setup form (each function/line is it's own form) uses [ManualControlSettings.ChannelGroups](https://github.com/d-ronin/dRonin/blob/next/shared/uavobjectdefinition/manualcontrolsettings.xml#L5) to decide what goes in the channel number dropdown but the RSSI function is passing in [ManualControlSettings.RssiType](https://github.com/d-ronin/dRonin/blob/next/shared/uavobjectdefinition/manualcontrolsettings.xml#L32). For example if None is selected (ManualControlSettings.RssiType.None = 0), it ends up using the [PWM number of channels (8)](https://github.com/d-ronin/dRonin/blob/next/ground/gcs/src/plugins/config/inputchannelform.cpp#L128) rather than 0 as it should (ManualControlSettings.ChannelGroups.PWM = 0)! Took me a while to figure out why I couldn't find the ADC code...

tracernz committed:
Config:Input: Use board-specific ADC names for RSSI

Also fixes broken assumption that ManualControlSettings.ChannelGroups == ManualControlSettings.RssiType

tracernz commented:
Fixed the RSSI input type bug and added custom ADC names to it.

tracernz commented:
Set FrSkyPWM RSSI channel count to 1.
```
<tracernz> hshred: q: for FrskyPWM RSSI, how many channels should be available in the input config form? is the channel number even used?
<hshred> tracernz: the channel number isn't used
```

mlyle commented:
Is that last comment a pending item, @tracernz?

tracernz commented:
Nope, done.

mlyle commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 23 of 24 files at r1, 4 of 4 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/d-ronin/dronin/449#-:-K8WvEFkL5WQHtQqasUv:1417088721)_

<!-- Sent from Reviewable.io -->
